### ISSUE TITLE
Some tests in GitContextTest are not testing functionality properly 

### ISSUE THREAD

**Issue by diiinesh** [devonfw/IDEasy#577]
The tests from #161 should be properly testing the fetchIfNeeded function and also the fetch functionality.

### Actual behavior
```
@Test
  public void testFetchIfNeededRecentFetchHead(@TempDir Path tempDir) throws IOException {
    // arrange
    String repoUrl = "https://github.com/test";
    String remoteName = "origin";
    List<String> errors = new ArrayList<>();
    List<String> outs = new ArrayList<>();
    IdeContext context = newGitContext(tempDir);
    GitContext gitContext = new GitContextImpl(context);

    Path gitDir = tempDir.resolve(".git");
    Files.createDirectories(gitDir);
    Path fetchHead = gitDir.resolve("FETCH_HEAD");
    Files.createFile(fetchHead);
    Files.setLastModifiedTime(fetchHead, FileTime.fromMillis(System.currentTimeMillis()));

    // act
    gitContext.fetchIfNeeded(repoUrl, remoteName, tempDir);

    // assert
    // Since FETCH_HEAD is recent, no fetch should occur
    assertThat(errors.isEmpty()).isTrue();
    assertThat(outs.isEmpty()).isTrue();
  }
```
This does not really test anything and should be rewritten such that the functionallity is tested properly.

**Comment by hohwille** [comment]
For this test mockito could be really helpful:
We could create `ProcessContext` as mock and test if it was actually invoked to call git or not.
Then we could test the case where the timestamp is old and git pull shall be triggered as well as a test where the git call is omitted since the timestamp is recent.
See also #581 that fixed a bug in this context. Such test would have revealed it.

**Comment by diiinesh** [comment]
This issue was already fixed with #585.
